src/components/CategoryDropdown/CategoryDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import "@/src/styles/index.scss";
+
+interface CategoryDropdownProps {
+  onSelectCategory: (category: string) => void;
+}
+
+const CategoryDropdown: React.FC<CategoryDropdownProps> = ({
+  onSelectCategory,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const toggleMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  const closeMenu = () => {
+    setIsMenuOpen(false);
+  };
+  const handleCategoryClick = (category: string) => {
+    onSelectCategory(category);
+    closeMenu();
+  };
+  return (
+    <div className="relative">
+      <button onClick={toggleMenu} className="flex items-center ">
+        <h2 className="text-xl font-bold mb-2">Category</h2>
+        <svg
```

Comment:
Instead of this svg code, could you make specific svg as icons with a proper name and then put it here? Cause I am reading this now and I have no idea what the icons look like unless I load the repo. You want to be able to just read the code

Better yet you guys can just select an icon library.